src/fireedge/src/modules/components/Forms/Vm/AttachNicForm/Steps/AdvancedOptions/schema.js

```diff
@@ -0,0 +1,653 @@
+    label: T.TransmissionQueue,
+    tooltip: T.OnlySupportedForVirtioDriver,
+    type: INPUT_TYPES.TEXT,
+    fieldProps: ([_, AUTO] = []) => ({
+      disabled: AUTO || hasAlias || isAlias,
+    }),
+    dependOf: [PCI_TYPE_NAME, 'AUTO_VIRTIO_QUEUES'],
+    value: (_, form) => {
+      if (form?.getValues(`advanced.AUTO_VIRTIO_QUEUES`) && form?.setValue) {
+        form?.setValue(`advanced.VIRTIO_QUEUES`, 'auto')
+      }
+    },
+    validation: string()
+      .trim()
+      .default(() => undefined),
+    htmlType: ([value, _] = []) =>
+      value !== NIC_HARDWARE.EMULATED ? INPUT_TYPES.HIDDEN : 'number',
+    grid: { md: 4.5 },
+  },
+  {
+    name: 'AUTO_VIRTIO_QUEUES',
+    label: T.Auto,
+    tooltip: T.AutoVirtioQueues,
+    type: INPUT_TYPES.SWITCH,
+    dependOf: PCI_TYPE_NAME,
+    htmlType: (value) =>
+      value !== NIC_HARDWARE.EMULATED ? INPUT_TYPES.HIDDEN : INPUT_TYPES.SWITCH,
+    validation: boolean()
+      .notRequired()
+      .default(() => false)
+      .afterSubmit(() => undefined),
+    grid: { md: 1.5 },
+  },
+  // PCI Passthrough Automatic mode fields
+  {
+    name: DEVICE_LIST,
+    label: T.DeviceName,
+    type: INPUT_TYPES.AUTOCOMPLETE,
+    optionsOnly: true,
+    values: () => {
+      const { data: hosts = [] } = HostAPI.useGetHostsAdminQuery()
+      const pciDevices = hosts
+        .map(getPciDevices)
+        .flat()
+        .reduce(
+          (currentPCIS, newDevice) =>
+            currentPCIS.some((pci) => pci.ADDRESS === newDevice.ADDRESS) // Filter out devices with the same address
+              ? currentPCIS
+              : [...currentPCIS, newDevice],
+          []
+        )
+
+      return arrayToOptions(pciDevices, {
+        getText: ({ DEVICE_NAME } = {}) => DEVICE_NAME,
+        getValue: transformPciToString,
+      })
+    },
+    validation: string()
+      .trim()
+      .default(() => {
+        const {
+          DEVICE: dataDevice = undefined,
+          VENDOR: dataVendor = undefined,
+          CLASS: dataClass = undefined,
+          PROFILES: dataProfile = undefined,
+        } = defaultData
+
+        return (
+          dataDevice &&
+          dataVendor &&
+          dataClass &&
+          [dataDevice, dataVendor, dataClass, dataProfile].join(';')
+        )
+      })
+      .afterSubmit(() => undefined),
+    dependOf: PCI_TYPE_NAME,
+    htmlType: (pciTypeValue) =>
+      pciTypeValue !== NIC_HARDWARE.PCI_PASSTHROUGH_AUTOMATIC &&
+      INPUT_TYPES.HIDDEN,
+    grid: { md: 3 },
+  },
+  {
+    name: VENDOR,
+    label: T.Vendor,
+    type: INPUT_TYPES.TEXT,
+    notOnHypervisors: [lxc],
+    dependOf: [PCI_TYPE_NAME, DEVICE_LIST],
+    watcher: fillPCIAtributes(VENDOR),
+    htmlType: (_, context) => {
+      const values = context?.getValues() || {}
+
+      return (
+        values?.advanced?.PCI_TYPE !== NIC_HARDWARE.PCI_PASSTHROUGH_AUTOMATIC &&
+        INPUT_TYPES.HIDDEN
+      )
+    },
+    validation: string()
+      .when('PCI_TYPE', (type, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).